source/adapters/cuda/program.cpp

```diff
@@ -483,7 +483,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramGetFunctionPointer(
     UR_CHECK_ERROR(Ret);
   if (Ret == CUDA_ERROR_NOT_FOUND) {
     *ppFunctionPointer = 0;
-    Result = UR_RESULT_ERROR_INVALID_FUNCTION_NAME;
+    Result = UR_RESULT_ERROR_FUNCTION_ADDRESS_NOT_AVAILABLE;
```

Comment:
Only just read @ldrumm s comment which is making the point a lot more succinctly: https://github.com/oneapi-src/unified-runtime/pull/1136#issuecomment-1883553654

Reply:
Thinking about this further. Maybe it is better to append e.g.:

```
 + void setPluginSpecificMessage(CUresult cu_res, UR_RESULT UR_res) {
- void setPluginSpecificMessage(CUresult cu_res) {
  const char *error_string;
  const char *error_name;
  cuGetErrorName(cu_res, &error_name);
  cuGetErrorString(cu_res, &error_string);
  char *message = (char *)malloc(strlen(error_string) + strlen(error_name) + 2);
  strcpy(message, error_name);
  strcat(message, "\n");
  strcat(message, error_string);

+ setErrorMessage(message, UR_res);
- setErrorMessage(message, UR_RESULT_ERROR_ADAPTER_SPECIFIC);
  free(message);
}
```
from https://github.com/oneapi-src/unified-runtime/blob/1cd402ead34a54459a6acb85777fbec105a178a0/source/adapters/cuda/common.cpp#L138

Then this gives the option of passing in a UR_RESULT (or whatever it is called) different from UR_RESULT_ERROR_ADAPTER_SPECIFIC (in such cases that there is a clarifying mapping as I described in my first comment).
Then you can
- preserve the native error
- clarify it by providing a UR mapping if one exists
- If one doesn't exist (which might be the case in this example from your code) then you can pass in UR_RESULT_ERROR_ADAPTER_SPECIFIC, possibly with an accompanying message.

If such a protocol were decided across backends, then documented, this would make it easier to prevent such errors as this example in the future, and provide a consistent error message framework for contributors to use.